thoth/glyph/cli.py

```diff
@@ -93,18 +103,107 @@ def generate(output: str) -> None:
     type=str,
-    default="-",
+    required=True,
     help="Commit message to be classified",
 )
 def classify(message: str) -> None:
     """Generate CHANGELOG entries from the current Git project."""
     _LOGGER.info("Hello, glyph!")
-    if message == "-":
-        click.echo("Please enter message")
-    else:
-        classifier = load_model("../../data/model_commits_v2_quant.bin") 
-        label = classifier.predict(message)
-        click.echo("Label : " + str(label[0][0])[8:])
+    _LOGGER.info("Model Path : " + DEFAULT_MODEL_PATH)
+    classifier = load_model(DEFAULT_MODEL_PATH) 
+    label = classifier.predict(message.lower())
+    click.echo("Label : " + str(label[0][0])[8:])
 
+@cli.command()
+@click.option(
+    "--path",
+    "-p",
+    type=str,
+    required=True,
+    help="Path to Git repository"
+)
+@click.option(
+    "--start",
+    type=str,
+    help="Starting date"
+)
+@click.option(
+    "--end",
+    type=str,
+    help="End date"
+)
+@click.option(
+    "--start_tag",
+    type=str,
+    help="Starting date"
+)
+@click.option(
+    "--end_tag",
+    type=str,
+    help="End date"
+)
+@click.option(
+    "--output",
+    type=str,
+    help="Generated output file"
+)
+@click.option(
+    "--model",
+    type=str,
+    help="Generated output file"
+)
+def classifyrepo(path: str, start: str, end: str, start_tag: str, end_tag:str, output: str, model:str) -> None:
+    if model is None:
+        model = DEFAULT_MODEL_PATH
+    _LOGGER.info("Model Path : " + model)
+    classifier = load_model(model)
+    start_time = 0
+    end_time = sys.maxsize
+
+    if start is not None:
+        start_time = int(time.mktime(datetime.datetime.strptime(start, "%Y-%m-%d").timetuple()))
+
+    if end is not None:
+        end_time = int(time.mktime(datetime.datetime.strptime(end, "%Y-%m-%d").timetuple()))
+
+    repo = Repository(os.path.join(path, ".git"))
+    orig_messages = classify_by_date(repo, start_time, end_time)
+
+    if start_tag is not None and end_tag is not None:
+        orig_messages = classify_by_tag(repo, start_tag, end_tag)
+
+    print(str(len(orig_messages)) + " commits classified")
+    df = pd.DataFrame(orig_messages, columns = ['message'])
+    df = df.replace('\n','', regex=True)
+    commits = list(df['message'].astype(str))
+    labels = classifier.predict(commits)
+    res = list(zip(*labels))
+    res_list = [x[0] for x in res]
+    lst2 = [item[0] for item in res_list]
+    df['labels_predicted'] = lst2
+    if output is None:
+        print(df)
+    else:
+        df.to_csv(output, sep='\t')
+
+def classify_by_date(repo, start_time, end_time):
+    _LOGGER.info("Classifying commits in the given date-range")
+    messages = []
+    for commit in repo.walk(repo.head.target, GIT_SORT_TOPOLOGICAL):
+        if(commit.commit_time > start_time and commit.commit_time < end_time):
+             messages.append(commit.message.lower())
+    return messages
+
+def classify_by_tag(repo, start_tag, end_tag):
```

Comment:
Nice! WE will need these in kebechet, so it can easily use thoth-glyph as a library. You can check thamos that acts as a CLI and as a library. The CLI part is just thin wrapper to parse command line options and pass it to library functions. See for example this part:

https://github.com/thoth-station/thamos/blob/025110516520899867ca8d1a30e39bcb270c23ba/thamos/cli.py#L377-L388

Note the library functions are present in the lib.py file and the cli.py just calls them.

Reply:
Now all functions are called inside basic CLI wrappers.
The functions can now be called independently as library functions or directly using the CLI.